src/declarations/stencil-public-compiler.ts

```diff
@@ -2192,30 +2192,268 @@ export type OutputTarget =
   | OutputTargetStats
   | OutputTargetDistTypes;
 
+/**
+ * Our custom configuration interface for generated caching Service Workers
+ * using the Workbox library (see https://developer.chrome.com/docs/workbox/).
+ *
+ * Although we are using Workbox we are unfortunately unable to depend on the
+ * published types for the library because they must be compiled using the
+ * `webworker` lib for TypeScript, which cannot be used at the same time as
+ * the `dom` lib. As a workaround the properties are copy-pasted here from
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c7b4dadae5b320ad1311a8f82242b8f2f41b7b8c/types/workbox-build/generate-sw.d.ts#L3
+ */
 export interface ServiceWorkerConfig {
-  // https://developers.google.com/web/tools/workbox/modules/workbox-build#full_generatesw_config
+  swSrc?: string;
   unregister?: boolean;
 
+  /**
+   * The local directory you wish to match `globPatterns` against. The path is relative to the current directory.
+   */
+  globDirectory?: string;
```

Comment:
It looks like we [updated this type from `string | string[]` ](https://github.com/ionic-team/stencil/pull/3813/files#diff-c396c2ecfcbdf4562ae518122990719c12b5ed9bd64908ef2142620f6d1440dbL2202)to `string` - this should probably be considered a breaking change since we're changing the public interface of this type.